homeassistant/components/senseme/switch.py

```diff
@@ -0,0 +1,95 @@
+"""Support for Big Ass Fans SenseME switch."""
+from __future__ import annotations
+
+from typing import Any
+
+from aiosenseme import SensemeFan
+
+from homeassistant import config_entries
+from homeassistant.components.switch import (
+    SwitchDeviceClass,
+    SwitchEntity,
+    SwitchEntityDescription,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import SensemeEntity
+
+FAN_SWITCHS = [
+    # Turning on sleep mode will disable Whoosh
+    SwitchEntityDescription(
+        key="sleep_mode",
+        name="Sleep Mode",
+    ),
+    SwitchEntityDescription(
+        key="motion_fan_auto",
+        name="Motion",
+    ),
+]
+
+FAN_LIGHT_SWITCHES = [
+    SwitchEntityDescription(
+        key="motion_light_auto",
+        name="Light Motion",
+    ),
+]
+
+LIGHT_SWITCHES = [
+    SwitchEntityDescription(
+        key="sleep_mode",
+        name="Sleep Mode",
+    ),
+    SwitchEntityDescription(
+        key="motion_light_auto",
+        name="Motion",
+    ),
+]
```

Comment:
I could do `lambda`s here for getting and setting the value, but it will significantly increase the size of this.

I'm on the fence if its a good idea on or not.

Reply:
I don't think I've seen lambda used for switch platform yet with entity descriptions atleast, just value_fn and some button press_action. I'm not a fan of dynamic set/getattr as it's less concise. maybe id opt to keep logic in method but check key and specifically get/set the attrs.

Reply:
I adjusted it to use a `value_fn` and a `set_fn`

Reply:
nice guess it's just docs now then?

Reply:
I think so.  Waiting for https://github.com/home-assistant/home-assistant.io/pull/21105

Reply:
@bdraco I normally leave docs to others but it was basic enough I approved and merged for you so you can start on this ones docs

Reply:
Thanks.  That one was a pretty simple one :)